### PR TITLE
Add SchoolDocument class for paying_for_college

### PR DESCRIPTION
**Pull request by cwdavies**
Adds the model for the paying_for_college document that we search on.
This will allow the creation and indexing of all schools in ES7.

---

## Additions

- `SchoolDocument` - Represents the data we load into elasticsearch for paying_for_college.

## How to test this PR

1. Inside the python container of our local docker stack run `cfgov/manage.py search_index --rebuild -f` and all schools are loaded into ES7 index `paying_for_college`. Can be confirmed via the kibana dashboard.

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

File: cfgov/paying_for_college/documents.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    city = fields.TextField()
+    state = fields.TextField()
+
+    def prepare_autocomplete(self, instance):
+        alias_strings = [a.alias for a in instance.alias_set.all()]
+        nickname_strings = [n.nickname for n in instance.nickname_set.all()]
+        return alias_strings + nickname_strings
+
+    def prepare_city(self, instance):
+        return instance.city
```
Comment (alexm118) [inline cfgov/paying_for_college/documents.py:42]: We can use a fields list instead of a prepare function if we aren't making any changes to a field.

https://django-elasticsearch-dsl.readthedocs.io/en/latest/quickstart.html

File: cfgov/paying_for_college/documents.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from django_elasticsearch_dsl import Document, fields
+from django_elasticsearch_dsl.registries import registry
+from elasticsearch_dsl import analyzer, token_filter, tokenizer
+
+from paying_for_college.models import School
+
+
+label_autocomplete = analyzer(
```
Comment (alexm118) [inline cfgov/paying_for_college/documents.py:8]: We shouldn't duplicate our analyzers and filters, maybe define a generic `elasticsearch_helpers.py` file in the search app.

File: cfgov/paying_for_college/tests/test_documents.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    def test_prepare(self):
+        school = School(school_id="999999",
+                        settlement_school="Example University",
+                        city="Example City", state="VA")
+        doc = SchoolDocument()
+
+        prepared_data = doc.prepare(school)
+        self.assertEqual(prepared_data, {
+            'autocomplete': doc.prepare_autocomplete(school),
```
Comment (alexm118) [inline cfgov/paying_for_college/tests/test_documents.py:17]: I don't think we should be calling `doc.prepare_*` in our test assertions for the expected results, we can hard code them, or reference them directly from the school object. This test would be more testing that the `prepare` function calls each defined fields `prepare` function, rather than what we want to test which is that the prepare function properly maps all fields from our django model to the document class.

**Review by alexm118 — COMMENTED** [review]
Just a couple suggestions and a fix or two.

File: cfgov/paying_for_college/tests/test_documents.py (hunk 3 above)
Comment (cwdavies) [reply cfgov/paying_for_college/tests/test_documents.py:17]: Removed doc.prepare_city and doc.prepare_state from test assertion.
The city and state are now referenced from the school object.

File: cfgov/ask_cfpb/documents.py [hunk 4]
```diff
@@ -1,36 +1,8 @@
 from django_elasticsearch_dsl import Document, fields
 from django_elasticsearch_dsl.registries import registry
```
Comment (higs4281) [inline cfgov/ask_cfpb/documents.py:2]: Should these changes be i a separate PR for ask_cfpb? (same issue in PR 6075)

Comment (cwdavies) [reply cfgov/ask_cfpb/documents.py:2]: Agree. I created a separate ask_cfpb PR: https://github.com/cfpb/consumerfinance.gov/pull/6076

File: cfgov/paying_for_college/documents.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+from paying_for_college.models import School
+from search.elasticsearch_helpers import label_autocomplete
+
+
+@registry.register_document
+class SchoolDocument(Document):
+
+    autocomplete = fields.TextField(analyzer=label_autocomplete)
+    text = fields.TextField(attr='settlement_school')
```
Comment (higs4281) [inline cfgov/paying_for_college/documents.py:12]: We should not be indexing the "settlement_school" field. This is for internal use and would server no search purpose.

The things that would be useful for search are:
- primary_alias (a property that needs to called in via "prepare" on an instance, since it's not a db column
- pk
- city
- state
- zip5

Comment (cwdavies) [reply cfgov/paying_for_college/documents.py]: Added zip5 and changed settlement_school to primary_alias

File: cfgov/paying_for_college/documents.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    autocomplete = fields.TextField(analyzer=label_autocomplete)
+    text = fields.TextField(attr='primary_alias', boost=10)
+    nicknames = fields.TextField()
+
+    def prepare_autocomplete(self, instance):
+        alias_strings = [a.alias for a in instance.alias_set.all()]
+        nickname_strings = [n.nickname for n in instance.nickname_set.all()]
+        return alias_strings + nickname_strings
+
+    def prepare_primary_alias(self, instance):
```
Comment (alexm118) [inline cfgov/paying_for_college/documents.py:20]: This can exist just within the fields list below.

Comment (alexm118) [reply cfgov/paying_for_college/documents.py:20]: Also - it looks like this prepare function isn't needed as you haven't defined a `primary_alias` field on this document. You are accessing the `primary_alias` data by indexing it within the `text` field.

Comment (cwdavies) [reply cfgov/paying_for_college/documents.py]: Removed prepare_primary_alias

**Review by alexm118 — APPROVED** [review]
LGTM
